projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/datajobs/DataJobsDeploymentController.java

```diff
@@ -112,39 +109,56 @@ public ResponseEntity<Void> deploymentPatch(
+    // TODO: deploymentId and mode not implemented
     if (jobsService.jobWithTeamExists(jobName, teamName)) {
-      // TODO: deploymentId and mode not implemented
-      List<DataJobDeploymentStatus> deployments = Collections.emptyList();
-      Optional<JobDeploymentStatus> jobDeploymentStatus = Optional.empty();
-      if (dataJobDeploymentPropertiesConfig.getReadDataSource().equals(ReadFrom.K8S)) {
-        jobDeploymentStatus = deploymentService.readDeployment(jobName.toLowerCase());
-      }
-      if (jobDeploymentStatus.isPresent()) {
-        deployments =
-            Arrays.asList(ToApiModelConverter.toDataJobDeploymentStatus(jobDeploymentStatus.get()));
-      }
-      return ResponseEntity.ok(deployments);
+      return deploymentAsList(jobName.toLowerCase());
     }
     return ResponseEntity.notFound().build();
   }
 
+  private ResponseEntity<List<DataJobDeploymentStatus>> deploymentAsList(String jobName) {
+    ResponseEntity<DataJobDeploymentStatus> jobDeploymentStatus;
+    if (dataJobDeploymentPropertiesConfig.getReadDataSource().equals(ReadFrom.DB)) {
+      jobDeploymentStatus = readFromDB(jobName);
+    } else {
+      jobDeploymentStatus = readFromK8S(jobName);
+    }
+    var response = Arrays.asList(jobDeploymentStatus.getBody());
+    return ResponseEntity.status(jobDeploymentStatus.getStatusCode()).body(response);
+  }
+
   @Override
   public ResponseEntity<DataJobDeploymentStatus> deploymentRead(
       String teamName, String jobName, String deploymentId) {
     if (jobsService.jobWithTeamExists(jobName, teamName)) {
-      // TODO: deploymentId are not implemented.
-      Optional<JobDeploymentStatus> jobDeploymentStatus = Optional.empty();
-      if (dataJobDeploymentPropertiesConfig.getReadDataSource().equals(ReadFrom.K8S)) {
-        jobDeploymentStatus = deploymentService.readDeployment(jobName.toLowerCase());
-      }
-      if (jobDeploymentStatus.isPresent()) {
-        return ResponseEntity.ok(
-            ToApiModelConverter.toDataJobDeploymentStatus(jobDeploymentStatus.get()));
+
+      if (dataJobDeploymentPropertiesConfig.getReadDataSource().equals(ReadFrom.DB)) {
+        return readFromDB(jobName.toLowerCase());
+      } else {
+        return readFromK8S(jobName.toLowerCase());
       }
     }
     return ResponseEntity.notFound().build();
   }
 
+  private ResponseEntity<DataJobDeploymentStatus> readFromK8S(String jobName) {
+    Optional<JobDeploymentStatus> jobDeploymentStatus =
+        deploymentService.readDeployment(jobName.toLowerCase());
+    if (jobDeploymentStatus.isPresent()) {
+      return ResponseEntity.ok(
+          ToApiModelConverter.toDataJobDeploymentStatus(jobDeploymentStatus.get()));
+    }
+    return ResponseEntity.notFound().build();
+  }
+
+  private ResponseEntity<DataJobDeploymentStatus> readFromDB(String dataJobName) {
+    var jobOptional = deploymentServiceV2.readDeploymentFromDB(dataJobName);
```

Comment:
I don't think the method called `readDeployment` returns `job`. Use better naming.